12451&13114 fix location ux review issues

ChasBoucher opened the pull request:
*Issue #:* /bcgov/entity#12451

*Description of changes:*

- This pr is bug fixes and style changes.
- The fixes required are detailed in ticket #12451 and #13114

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

ChasBoucher committed:
12451-Fix Group spacing

ChasBoucher committed:
12451 Fix Stepper Validation

ChasBoucher committed:
12451-Fix issue with group header and labels

ChasBoucher committed:
12451-Fix issues with review form

ChasBoucher committed:
12451-Fix styling issues

ChasBoucher committed:
12451-Fix more styling issues

ChasBoucher committed:
12451-Final changes for ticket 12451

ChasBoucher committed:
12451-Fix display of group interest type

ChasBoucher committed:
12451-Fix test

ChasBoucher committed:
12451-Fix error for group allocation

cameron-eyds commented on ppr-ui/src/components/mhrRegistration/HomeOwners/FractionalOwnership.vue:
```diff
@@ -122,11 +131,15 @@ export default defineComponent({
 </script>
 
 <style lang="scss" scoped>
+.backgroundWhite {
```
`.background-white` 
Kebab case 👍

ChasBoucher replied:
fixed.  old habits die hard...

ChasBoucher replied:
this is a weird one... I changed to background-white and class and style bindings stopped working. ie would not compile..  I changed to backgroundwhite and it works...

cameron-eyds commented on ppr-ui/src/components/mhrRegistration/HomeOwners/FractionalOwnership.vue:
```diff
@@ -122,11 +131,15 @@ export default defineComponent({
 
 <style lang="scss" scoped>
+.backgroundWhite {
+  background:white;
```
Missing a space here

ChasBoucher replied:
fixed

cameron-eyds commented on ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnerGroups.vue:
```diff
@@ -150,14 +150,19 @@ export default defineComponent({
 
+.pad-top {
+  padding-top: 0px;
+}
```
You could get away with a vuetify helper here, `pt-0`

ChasBoucher replied:
nice... I need to get more familiar with those..

cameron-eyds replied:
For sure! They are quite helpful, especially when making those small adjustments.

cameron-eyds commented on ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnerGroups.vue:
```diff
@@ -150,14 +150,19 @@ export default defineComponent({
   }
+  .v-application .my-8 {
+    margin-bottom: 0px;
+  }
```
This is a little confusing as `my-8` is essentially the same as 32px Top and Bottom. 
Why not change `my-8` to `mt-8` and if the margin-bottom needs adjusting apply a `mb-0` class.

ChasBoucher replied:
fixed.  I like helpers...

cameron-eyds commented on ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnersTable.vue:
```diff
@@ -41,10 +41,10 @@
+                  {{ row.item.individualName.first }}
+                  {{ row.item.individualName.middle }}
+                  {{ row.item.individualName.last }}
               </div>
```
Looks like additional spacing here

ChasBoucher replied:
fixed

ChasBoucher replied:
fixed

cameron-eyds commented on ppr-ui/src/composables/mhrRegistration/home-owner.ts:
```diff
@@ -62,7 +62,7 @@ export function useHomeOwners (isPerson: boolean = false, isEditMode: boolean =
       // At leas one group showing with one or more owners
       return HomeTenancyTypes.COMMON
-    } else if (numOfOwners === 1) {
+    } else if (numOfOwners === 1 && getMhrRegistrationHomeOwners.value[0].address !== undefined) {
```
If we are looking for just proof the property exists or is initialized, does `!!getMhrRegistrationHomeOwners.value[0].address` work?

Also, should we optionally chain `getMhrRegistrationHomeOwners.value[0]` or is this always initialized?

ChasBoucher replied:
Good comment.. getMhrRegistrationHomeOwners.value[0] is always initialized when a new group is created..

cameron-eyds commented on ppr-ui/src/composables/mhrRegistration/home-owner.ts:
```diff
@@ -94,11 +94,18 @@ export function useHomeOwners (isPerson: boolean = false, isEditMode: boolean =
+  }
+  const getNumberOfGroups = (): boolean => {
+    return !getMhrRegistrationHomeOwnerGroups.value || getMhrRegistrationHomeOwnerGroups.value.length < 2
+  }
```
The property name is a little misleading, `getNumberOfGroups` to return a `boolean`. Maybe we could better describe what this expression is doing in the name. ie `hasMinimumGroups` or whatever

ChasBoucher replied:
good catch

cameron-eyds commented on ppr-ui/src/composables/mhrRegistration/home-owner.ts:
```diff
@@ -284,22 +291,32 @@ export function useHomeOwners (isPerson: boolean = false, isEditMode: boolean =
+    ((getMhrRegistrationHomeOwnerGroups.value.length === 1 && !getMhrRegistrationHomeOwners.value.address) &&
+      getMhrRegistrationHomeOwnerGroups.value[0].owners.length <= 1 && getMhrRegistrationHomeOwnerGroups.value[0].interestNumerator === null)
+  }
+
```
A small comment describing this chain of operators would be useful to other devs i think

ChasBoucher replied:
Good suggestion..

cameron-eyds commented on ppr-ui/src/views/newMhrRegistration/HomeOwners.vue:
```diff
@@ -88,9 +88,11 @@
-        <span class="generic-label">Home Tenancy Type: </span>{{ getHomeTenancyType() }}
+        <div><span class="generic-label">Home Tenancy Type: </span>{{ homeTenancyType }}<span v-show="showGroups"><span v-show="ownershipAllocation.hasMinimumGroupsError" class="error-text fs-14 ml-3"
+            >Must include more than one group of owners</span
+          ></span></div>
```
Could we clean this up a bit to make it a little more readable? 
Also, is a `div` wrapper required here?

ChasBoucher replied:
I cleaned up the code, but I did'nt think the div was necessary. I used a span because of the added error logic...

cameron-eyds commented on ppr-ui/src/views/newMhrRegistration/HomeOwners.vue:
```diff
@@ -88,9 +88,11 @@
+          ></span></div>
         <div v-show="showGroups">
-          <span class="generic-label">Total Ownership Allocated: </span>{{ ownershipAllocation.totalAllocation }}
+          <span class="generic-label">Total Ownership Allocated:</span><span> {{ interestType }} </span>{{ ownershipAllocation.totalAllocation }}
```
A newline never hurts for readability, with multiple tags/computed values etc.

ChasBoucher replied:
agreed.. fixed..

cameron-eyds commented on ppr-ui/src/views/newMhrRegistration/HomeOwners.vue:
```diff
@@ -178,7 +183,10 @@ export default defineComponent({
+      ),
+      isValidGroups: computed(() => { return getNumberOfGroups() }),
+      homeTenancyType: computed(() => { return getHomeTenancyType() }),
+      interestType: computed(() => { return getInterestString() })
```
Are these required? 
Seems to be an alias for the function expressions from the composable or was the intention to do this so the `setup()` return was lessened?

ChasBoucher replied:
I only did this for readability...

dimak1 replied:
This is confusing: `isValidGroups: computed(() => { return getNumberOfGroups() })`. In the name, it sounds like a _boolean_ but yet the computed returns a _number_ of groups.

ChasBoucher replied:
I made this change according to an earlier comment by Cameron.. good catch though... I changed to hasMinimumGroups.

cameron-eyds reviewed (commented):
Great work, a lot of good stuff here. Few questions and comments.

dimak1 commented:
Hi @ChasBoucher, there are a quite number of changes in this PR, and I wanted to understand them so I checked out the PR. Right away I see a bug - when we delete an Owner, the group header still shows as Owners: 1. I suspect this because the `TableGroupHeader` items prop had a helper method removed (`hasActualOwners`).

<img width="1071" alt="Screen Shot 2022-09-19 at 10 07 14 AM" src="https://user-images.githubusercontent.com/2333290/191074466-11f1543e-78d2-4802-b5ca-9ddf98a07332.png">

dimak1 commented on ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnersTable.vue:
```diff
@@ -100,14 +103,14 @@
         </tr>
       </template>
       <template v-slot:no-data>
-        <div class="my-6">
```
We need this here, otherwise it will show a default empty slot. 

<img width="994" alt="Screen Shot 2022-09-19 at 10 16 50 AM" src="https://user-images.githubusercontent.com/2333290/191075301-d3f8a9c2-2b4c-4a1f-ad8c-b7e1cfa29610.png">

ChasBoucher replied:
Done...

dimak1 commented on ppr-ui/src/components/mhrRegistration/HomeOwners/FractionalOwnership.vue:
```diff
@@ -12,14 +12,17 @@
           :id="`fraction-amount-group-${groupId}`"
           label="Amount Owned by this Group"
-          filled
+          :filled="!editingGroup"
```
These two changes cause the following:

<img width="980" alt="Screen Shot 2022-09-19 at 10 44 27 AM" src="https://user-images.githubusercontent.com/2333290/191081522-0a8b6a29-28c0-4ceb-85ab-ff79f4c8dbf9.png">

dimak1 replied:
@ChasBoucher white background (as per design specs) could be achieved with the following (works on my local):

```
<v-text-field
   ...
   filled
   class="background-white"
   ...
/>
```


```
  .theme--light.v-text-field--filled.background-white > .v-input__control > .v-input__slot {
    background: white;
  }
```

ChasBoucher replied:
There is a styling issue.. The requirement is correct.. I may need assistance fixing this one...

ChasBoucher replied:
Fixed issue with Dima... Works!

cameron-eyds reviewed (approved):
Code looks good, thanks for the updates. 
I would rely on Dimas familiarity here to give the final 👍

ChasBoucher commented:
I have addressed the issues, but am having difficulty with the styling on Fractional ownership in edit mode.  The background of the text should be white when a groups fractional ownership is being edited..

ChasBoucher committed:
12451-Fix review issues

dimak1 commented:
> I have addressed the issues, but am having difficulty with the styling on Fractional ownership in edit mode. The background of the text should be white when a groups fractional ownership is being edited..

It can be fixed with the above code in my comment. Here is a screenshot: 

<img width="712" alt="Screen Shot 2022-09-19 at 12 22 03 PM" src="https://user-images.githubusercontent.com/2333290/191098513-85e8c0c6-31b9-4498-92d7-891e8f03c396.png">

ChasBoucher committed:
12451-Fix styling issue with Fractional ownership

ChasBoucher commented:
Fixed all issues in tickets #12451 and #13114.

Once merged tickets 13567(may be fixed allready), #13534 and other related tickets will be picked up by @dimak1.

dimak1 commented on ppr-ui/src/components/mhrRegistration/HomeOwners/FractionalOwnership.vue:
```diff
@@ -77,6 +81,10 @@ export default defineComponent({
       type: Object,
       default: null,
       required: true
+    },
```
Could be removed.

ChasBoucher replied:
Removed

ChasBoucher replied:
Rmoved

dimak1 commented on ppr-ui/src/composables/mhrRegistration/home-owner.ts:
```diff
@@ -284,22 +292,46 @@ export function useHomeOwners (isPerson: boolean = false, isEditMode: boolean =
+      When all owners are deleted 1 owner still exists without
+      and address and the interestinumerator will be blank... )
+  */
+  const showGroupHeading = (): Boolean => {
```
Not sure we need this? Yes, the groups are initialized with '1', but the initial value for `showGroups` is false. It will be set to true once the user decides to add a Group. The only time the groups length will be 0 is when we delete all groups and therefore reset `showGroups` to false.

ChasBoucher replied:
The reason I put in this logic, is because is because when a person was deleted, it added a  group heading and displayed Tenants in Common.. This logic fixed that issue...

eg.  
![image](https://user-images.githubusercontent.com/107890689/191144402-e37c49dc-6f63-42d8-b9d4-95598a7c82bf.png)

dimak1 replied:
Is there a way to simplify it? Also, `getMhrRegistrationHomeOwners` is an array, so this `getMhrRegistrationHomeOwners.value.address` will not work...

ChasBoucher replied:
Worked with Dima to rework logic.  fixed.

dimak1 commented:
@ChasBoucher looks much better after addressing the comments. I added few more.

ChasBoucher committed:
12451-Fix Pr issues and bug fix

ChasBoucher committed:
12451-Fix group error

ChasBoucher committed:
12451-Cleanup

ChasBoucher committed:
12451-Remove unused code.
